### ISSUE TITLE
Integrate with Basis Universal format

### ISSUE THREAD

**Issue by kvark** [gfx-rs/wgpu-rs#491]
When we start supporting #141 , we'll have this problem that no compressed format is supported everywhere. It would be nice to still provide *some* way to have compressed textures to the user, way that would work everywhere. [Basis Universal](https://github.com/BinomialLLC/basis_universal) is a good candidate to solve this. There is even a crate in rust that implements it - https://github.com/JakubValtar/basisu_rs (cc @JakubValtar). We should explore an opportunity to depend on that crate, optionally, so that we can hide the differences between compressed formats at wgpu-rs level.

**Comment by JakubValtar** [comment]
Hey, popping in since I was mentioned.

`basisu_rs` is very experimental and can't do much yet (like transcoding) or even do it properly.

If you want to add support for Basis Universal, please use the [official transcoder](https://github.com/BinomialLLC/basis_universal), which is production-ready.

**Comment by kvark** [comment]
Thanks for the update @JakubValtar ! We are very hesitant to add non-Rust dependencies here, so we'd love to see your work maturing some day :)

**Comment by JakubValtar** [comment]
Yeah, I understand. I'm working on it when I have time, but it will take a while. I'm not sure if I'll ever come close to the official implementation.

The official transcoder has been thoroughly fuzzed, tested, and optimized. As far as I know, it is being used by all the projects supporting Basis. I don't think there will be any higher quality alternative anytime soon – if ever.
